analyzer/tools/build-logger/src/logger.c

```diff
@@ -0,0 +1,103 @@
+/**
+ * -------------------------------------------------------------------------
+ *                     The CodeChecker Infrastructure
+ *   This file is distributed under the University of Illinois Open Source
+ *   License. See LICENSE.TXT for details.
+ * -------------------------------------------------------------------------
+ */
+
+#include "logger.h"
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+
+/**
+ * Returns formatted current time.
+ * 
+ * @param buff_ an output buffer (non null).
+ * @return anways returns buff_.
+ */
+void getCurrentTime(char* buff_)
+{
+  time_t rawtime;
+  struct tm* timeinfo;
+
+  time(&rawtime);
+  timeinfo = localtime(&rawtime);
+
+  strftime(buff_, 26, "%Y-%m-%d %H:%M:%S", timeinfo);
+
+  return buff_;
+}
+
+void logPrint(char* logLevel_, char* fileName_, int line_, char *fmt_,...)
+{
+  const char* debugFile = getenv("CC_LOGGER_DEBUG_FILE");
+  if (!debugFile)
+  {
+    return 0;
+  }
+
+  FILE* fp = fopen(debugFile, "a");
```

Comment:
I think it's not a problem that the file is opened at each build action, because the logger itself does the same when appending the compilation database. However, I think we should use the same locking system in order to avoid issues in multithreaded logging. See: https://github.com/Ericsson/codechecker/blob/b66c0fb116dd098061528af64eeec26deb8ca10b/analyzer/tools/build-logger/src/ldlogger-logger.c#L242-L266